transcom/mymove
# B-20229 refactor fuel pricer

[B-20229](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A964589)

# Summary

Update FRA weekly effective period
 # Background
the app pulls the diesel prices every day looking for newly published prices
EIA publishes new prices generally every Monday, before this change they would just be effective until the next publication.
This change makes it so that when new diesel prices are published they are only valid from tueday( or the day after they are published0 through the following Monday 

# to test
1. run `make tasks_save_ghc_fuel_price_data` to get the latest pricing data for your test
2. create a new move
3. approve move
4. access prime and create a payment request for fuel surcharge 
5. login as TIO - review the payment request
6. validate that it selected the correct price based on the actual pickup date being between the effective and End dates in the DB GHC_DIESEL_FUEL_PRICES

- may want to manually enter a few extra rows verify it works and making the actual pickup date land on the effecting and end dates 
<img width="1070" alt="Screenshot 2024-06-07 at 2 26 11 PM" src="https://github.com/transcom/mymove/assets/136509766/bd89f7ae-cde6-47c2-8211-cc8333dc15c2">
<img width="1070" alt="Screenshot 2024-06-07 at 2 57 05 PM" src="https://github.com/transcom/mymove/assets/136509766/e5ffb0dc-4792-4b9b-9ad0-50b157f595bc">


<img width="1277" alt="image" src="https://github.com/transcom/mymove/assets/136509766/8f04c520-a91e-4a0a-99bd-2ce95aa9f80d">
